drbert/data/dataset_readers.py

```diff
@@ -0,0 +1,242 @@
+    def __init__(self, path, partitions, tokenizer, format='json', batch_sizes=None, maxlen=512,
+                 lower_case=False):
+        super(DocumentClassificationDatasetReader, self).__init__(
+            path, partitions, tokenizer, batch_sizes, maxlen, lower_case,
+            # Sort examples according to length of documents
+            sort_key=lambda x: len(x.text)
+        )
+
+    def textual_to_iterator(self):
+        """Does whatever tokenization or processing is necessary to go from textual input to
+        iterators for training and evaluation for a document classification dataset. The data
+        must be in JSON lines format (http://jsonlines.org/).
+
+        Returns:
+            A tuple of `torchtext.data.iterator.BucketIterator` objects, one for each partition
+            is `self.partitions`.
+        """
+        # To use the tokenizer from Transformers, we need to do three things:
+        # 1. Set use_vocab=False
+        # 2. Set tokenizer=tokenizer.encode
+        # 3. Provide indices for all special tokens (init, eos, pad & unk)
+        TEXT = data.Field(
+            use_vocab=False,
+            init_token=self.tokenizer.cls_token_id,
+            eos_token=self.tokenizer.sep_token_id,
+            fix_length=self.maxlen,
+            tokenize=self.tokenizer.encode,
+            batch_first=True,
+            pad_token=self.tokenizer.pad_token_id,
+            unk_token=self.tokenizer.unk_token_id
+        )
+        LABEL = data.LabelField(fix_length=self.maxlen)
+
+        fields = {'text': ('text', TEXT), 'label': ('label', LABEL)}
+
+        splits, iterators = super().textual_to_iterator(fields)
+
+        # Finally, build the vocab. This "numericalizes" the field.
+        # Assume the first split is 'train'.
+        LABEL.build_vocab(splits[0])
+
+        return iterators
+
+
+class SequenceLabellingDatasetReader(DatasetReader):
+    """The parent class for all dataset readers.
+
+    Args:
+        path (str): Common prefix of the splits’ file paths.
+        partitions (dict): A dictionary keyed by partition ('train', 'test', 'validation')
+            containing the suffix to add to `path` for that partition.
+        tokenizer (transformers.PreTrainedTokenizer): The function used to tokenize strings into
+            sequential examples.
+        format (str): The format of the data file. One of “CSV”, “TSV”, or “JSON” (case-insensitive).
+        batch_sizes (tuple): Optional, tuple of batch sizes to use for the different splits, or None
+            to use the same batch_size for all splits. Defaults to None.
+        maxlen (int): Optional, a fixed length that all examples using this field will be padded to,
+            for flexible sequence lengths. Default: 512.
+        lower_case: (bool): Optional, True if text should be lower cased, False if not. Defaults to
+            False.
+    """
+    def __init__(self, path, partitions, tokenizer, format='csv', batch_sizes=None, maxlen=512,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Redefining built-in 'format'](https://app.codacy.com/app/JohnGiorgi/drbert/pullRequest?prid=4302662)